addons/im_livechat/models/ir_websocket.py

```diff
@@ -0,0 +1,20 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+from odoo.addons.im_livechat.tools.misc import force_guest_env
+from odoo.http import request
+from odoo.addons.bus.websocket import wsrequest
+
+class IrWebsocket(models.AbstractModel):
+    _inherit = "ir.websocket"
+
+    def _build_bus_channel_list(self, channels):
+        guest_channel = next(
+            (c for c in channels if isinstance(c, str) and c.startswith("mail.guest_")),
+            None,
+        )
+        if guest_channel:
+            channels = [c for c in channels if c != guest_channel]
+            force_guest_env(guest_channel.split("_")[1])
+            self = self.with_context(**(request or wsrequest).env.context)
+        return super()._build_bus_channel_list(channels)
```

Comment:
I think you need to do a py2 style `super()` here if you want that `super()` use the reassigned `self`
```suggestion
            self = self.with_context(**(request or wsrequest).env.context)
        return super(IrWebsocket, self)._build_bus_channel_list(channels)
```

Reply:
It works as intended i.e. the context used in the super method is up to date, I guess this is not required afterall? :man_shrugging: